datastr/dict.go

```diff
@@ -17,16 +18,23 @@ const (
 type Dict struct {
 	hashTables [2]*HashTable
 	rehashidx  int
+	rwmux sync.RWMutex
```

Comment:
All these properties are adding entropy to the Dict, violating the single responsability principle. The Dict must own and handle the hashtables and is not responsible for generating random bytes and splitting them.
The concurrency can be handle at operation level.

Reply:
ok

Reply:
> and is not responsible for generating random bytes and splitting them.

hum yeah but before it did split the static hashkey every request to get key0 and key1. 
these are now fixed and carried in the struct. and can be accessed with .key0 .key1 from inside

Reply:
i'll patchout the logger stuff and we'll see what remains